worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java

```diff
@@ -336,16 +337,14 @@ public void switchServingState() {
     }
     switch (servingState) {
       case PUSH_PAUSED:
-        if (canResumeByPinnedMemory()) {
-          resumeByPinnedMemory(servingState);
-        } else {
+        if (!tryResumeByPinnedMemory(servingState, lastState)) {
           pausePushDataCounter.increment();
           if (lastState == ServingState.PUSH_AND_REPLICATE_PAUSED) {
-            logger.info("Trigger action: RESUME REPLICATE");
             resumeReplicate();
           } else {
             logger.info("Trigger action: PAUSE PUSH");
             pausePushDataStartTime = System.currentTimeMillis();
+            resumingByPinnedMemory = false;
```

Comment:
resumingByPinnedMemory need also change to false when servingState changes to NONE_PAUSED state